c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2029,6 +2578,17 @@ class DeviceCachingAllocator {
         total_allocated_memory + size > allowed_memory_maximum) {
       p.err = cudaErrorMemoryAllocation;
       return false;
+    } else if (
+        CachingAllocatorConfig::expandable_segments() &&
+        !p.pool->owner_PrivatePool) {
```

Comment:
Add comment that this is because expandable segments dont work with checkpointing ?